pkg/services/ngalert/store/alertmanager.go

```diff
@@ -132,6 +146,48 @@ func (st *DBstore) UpdateAlertmanagerConfiguration(ctx context.Context, cmd *mod
 	})
 }
 
+// MarkConfigurationAsApplied sets the `last_applied` field of the last config with the given hash to the current UNIX timestamp.
+func (st *DBstore) MarkConfigurationAsApplied(ctx context.Context, cmd *models.MarkConfigurationAsAppliedCmd) error {
+	return st.SQLStore.WithTransactionalDbSession(ctx, func(sess *db.Session) error {
+		update := map[string]interface{}{"last_applied": time.Now().UTC().Unix()}
+		rowsAffected, err := sess.Table("alert_configuration_history").
+			Desc("id").
+			Limit(1).
+			Where("org_id = ? AND configuration_hash = ?", cmd.OrgID, cmd.ConfigurationHash).
+			Cols("last_applied").
+			Update(&update)
+
+		if err != nil {
+			return err
+		}
+
+		// We should be updating 1 record max.
+		if rowsAffected > 1 {
+			return fmt.Errorf("update statement affected %d alert configuration history records", rowsAffected)
+		}
+
+		return nil
+	})
+}
+
+// GetAppliedConfigurations returns all configurations that have been marked as applied, ordered newest -> oldest.
```

Comment:
```suggestion
// GetAppliedConfigurations returns all configurations that have been marked as applied, ordered newest -> oldest by id.
```
Nit: Newest->oldest by applied date, or by ID, may not be exactly the same thing

Reply:
System timestamps are unfortunately not reliable and not guaranteed to be monotonically increasing

Reply:
Ohh you're totally right, Martin Kleppmann would be dissapointed 😬